x-pack/plugins/observability_solution/logs_explorer/common/data_views/models/data_view_descriptor.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DataViewListItem } from '@kbn/data-views-plugin/common';
+import { TIMESTAMP_FIELD } from '../../constants';
+import { DataViewSpecWithId } from '../../dataset_selection';
+import { DataViewDescriptorType } from '../types';
+import { buildIndexPatternRegExp } from '../utils';
+
+type Whitelist = Array<string | RegExp>;
+
+const LOGS_WHITELIST: Whitelist = [
+  buildIndexPatternRegExp(['logs', 'auditbeat', 'filebeat', 'winbeat']),
+  // Add more strings or regex patterns as needed
+];
+
+export class DataViewDescriptor {
+  id: DataViewDescriptorType['id'];
+  dataType: DataViewDescriptorType['dataType'];
+  name: DataViewDescriptorType['name'];
+  namespaces: DataViewDescriptorType['namespaces'];
+  title: DataViewDescriptorType['title'];
+  type: DataViewDescriptorType['type'];
+
+  private constructor(dataViewDescriptor: DataViewDescriptorType) {
+    this.id = dataViewDescriptor.id;
+    this.dataType = dataViewDescriptor.dataType;
+    this.name = dataViewDescriptor.name;
+    this.namespaces = dataViewDescriptor.namespaces;
+    this.title = dataViewDescriptor.title;
+    this.type = dataViewDescriptor.type;
+  }
+
+  getFullTitle() {
+    return this.name;
+  }
+
+  toDataviewSpec(): DataViewSpecWithId {
+    return {
+      id: this.id,
+      name: this.name,
+      timeFieldName: TIMESTAMP_FIELD,
+      title: this.title,
+    };
+  }
+
+  toPlain() {
+    return {
+      id: this.id,
+      dataType: this.dataType,
+      name: this.name,
+      title: this.title,
+    };
+  }
+
+  public static create(dataViewListItem: DataViewListItem) {
+    const name = dataViewListItem.name ?? dataViewListItem.title;
+    const dataType = DataViewDescriptor.#extractDataType(dataViewListItem.title);
+
+    return new DataViewDescriptor({ ...dataViewListItem, dataType, name });
+  }
+
+  static #extractDataType(title: string): DataViewDescriptorType['dataType'] {
```

Comment:
**note**: I made this method a private one to prevent its usage in rendering loops or at render time, as it can be an expensive invocation due to its implementation. If is necessary to use the instance dataType, the utility methods or the property itself can be directly used.